internal/topology/variables/cluster_variable_validation.go

```diff
@@ -151,34 +167,66 @@ func ValidateClusterVariable(value *clusterv1.ClusterVariable, definition *clust
 		return err
 	}
 
-	return validateUnknownFields(fldPath, value, variableValue, apiExtensionsSchema)
-}
-
-// validateUnknownFields validates the given variableValue for unknown fields.
-// This func returns an error if there are variable fields in variableValue that are not defined in
-// variableSchema and if x-kubernetes-preserve-unknown-fields is not set.
-func validateUnknownFields(fldPath *field.Path, clusterVariable *clusterv1.ClusterVariable, variableValue interface{}, variableSchema *apiextensions.JSONSchemaProps) field.ErrorList {
 	// Structural schema pruning does not work with scalar values,
 	// so we wrap the schema and the variable in objects.
 	// <variable-name>: <variable-value>
 	wrappedVariable := map[string]interface{}{
-		clusterVariable.Name: variableValue,
+		definition.Name: variableValue,
```

Comment:
Q: Why was this changed from clusterVariable/value.Name to definition.Name?

Reply:
This will be equivalent, but I've changed it back to `value.Name`

Reply:
Sounds good. Was mostly wondering if there was a reason why it had to be changed that I'm missing